pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,100 @@
 Contains the MPSPrep template.
 """
 
+import numpy as np
+
 import pennylane as qml
 from pennylane.operation import Operation
 from pennylane.wires import Wires
 
 
+def right_canonicalize_mps(mps):
+    """Transform an MPS into a right-canonical MPS.
+
+    Args:
+        mps (list[TensorLike]): List of tensors representing the MPS.
+
+    Returns:
+        A list of tensors representing the MPS in right-canonical form.
+
+    .. seealso:: :class:`~.MPSPrep`.
+
+    """
+
+    is_right_canonical = True
+    d_shapes = []
+    for tensor in mps[1:-1]:
+        # Right-canonical definition
+        d_shapes += tensor.shape
+        M = qml.math.tensordot(tensor, tensor.conj(), axes=([1, 2], [1, 2]))
+        if not qml.math.allclose(M, qml.math.eye(tensor.shape[0])):
+            is_right_canonical = False
+
```

Comment:
Can break here?

Reply:
I decided to do it like this so I can use this loop also to calculate the shapes:

```
        d_shapes += tensor.shape
```